### PR TITLE
Python 3.10 support

### PR DESCRIPTION
**Pull request by cheesycod**
## What do these changes do? Adds Python 3.10 support

Python 3.10 removed loop from asyncio calls. My patch might not be the way to fix this however

## Checklist

- [X ] I think the code is well written
- [ ] Unit tests for the changes exist
- [X] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt` 

I do not wish to be added to CONTRIBUTORS.txt due to privacy reasons

  * The format is `<Name> <Surname>`.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the [`CHANGES/`](../tree/master/CHANGES) folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example:
   `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by cheesycod** [comment]
@hotsyk Can you review this?

**Comment by seandstewart** [comment]
Hey @cheesycod -

Thanks for the submission! Could you also go ahead and add `3.10-dev` to the python version matrix in our CI?

**Comment by cheesycod** [comment]
@seandstewart I think I did it but not sure if that’s how you are supposed to add it

**Comment by cheesycod** [comment]
@samuelcolvin the uvloop checks are failing but that’s unrelated to aioredis (and needs a recythonize)

**Comment by samuelcolvin** [comment]
Maybe I'm missing something, but not sure why this relates to me?

**Comment by Andrew-Chen-Wang** [comment]
@samuelcolvin may have been caught in an accidental autocomplete issue (i.e. when you type in \@ etc. etc.)....

File: aioredis/connection.py [hunk 1]
```diff
@@ -679,7 +679,7 @@ async def _connect(self):
         """Create a TCP socket connection"""
         async with async_timeout.timeout(self.socket_connect_timeout):
             reader, writer = await asyncio.open_connection(
-                host=self.host, port=self.port, ssl=self.ssl_context, loop=self._loop
+                host=self.host, port=self.port, ssl=self.ssl_context
```
Comment (Andrew-Chen-Wang) [inline aioredis/connection.py:682]: Versions before Python 3.10 expect the loop argument to work though. Find a better way to input arguments without causing any warnings to appear but to still have compatibility with Python 3.9 and below

**Review by Andrew-Chen-Wang — CHANGES_REQUESTED** [review]

File: aioredis/connection.py (hunk 1 above)
Comment (cheesycod) [reply aioredis/connection.py:682]: I do not think python 3.9 or 3.8 iirc it was only for older python versions

Comment (cheesycod) [reply aioredis/connection.py:682]: @Andrew-Chen-Wang yeah, it was only for older python: https://stackoverflow.com/questions/60312374/what-are-all-these-deprecated-loop-parameters-in-asyncio

**Comment by abrookins** [comment]
@cheesycod Could you make one last change to this PR, which is to add a file in the CHANGES/ directory describing the change?

**Comment by abrookins** [comment]
@cheesycod Shoot - now the linter is complaining. If you run `make lint`, the linter should automatically reformat any problems it finds, then you can commit those and it should pass. _Should_. 🤞

**Comment by cheesycod** [comment]
This issue is no longer valid as of https://github.com/aio-libs/aioredis-py/commit/3974143dcdc615210f0934d7dc18f63eac9c65a4#diff-eebfcf75e78ea2114a6e6019304b6e5709d414bb1e98375a0beafcd593430231

**Comment by cheesycod** [comment]
Sorry, meant https://github.com/aio-libs/aioredis-py/commit/9f161edec93e151efc42d1983c3feef3a0178969#diff-eebfcf75e78ea2114a6e6019304b6e5709d414bb1e98375a0beafcd593430231

**Pull request closed without merging**